test_spack.py

```diff
@@ -284,44 +284,29 @@ class GridToolsTest(TestCase):
 class IconTest(TestCase):
     package_name = 'icon'
     depends_on = {'serialbox', 'eccodes', 'claw'}
-    machines = all_machines
+    machines = {'daint'}
 
     def test_install_nwp_gpu_nvidia(self):
         # So we can make sure ICON-NWP (OpenACC) devs can compile (mimicks Buildbot for Tsa)
-        if machine == 'tsa':
-            self.Srun(
-                'spack install --show-log-on-error icon@nwp%pgi icon_target=gpu +claw +eccodes +ocean'
-            )
-        else:
-            self.Srun(
-                'spack install --show-log-on-error icon@nwp%nvhpc icon_target=gpu +claw +eccodes +ocean'
-            )
+        self.Srun(
+            'spack install --show-log-on-error icon@nwp%nvhpc icon_target=gpu +claw +eccodes +ocean'
+        )
 
     def test_install_nwp_cpu_nvidia(self):
         # So we can make sure ICON-NWP (OpenACC) devs can compile (mimicks Buildbot for Tsa)
-        if machine == 'tsa':
-            self.Srun(
-                'spack install --show-log-on-error icon@nwp%pgi icon_target=cpu serialize_mode=create +eccodes +ocean'
-            )
-        else:
-            self.Srun(
-                'spack install --show-log-on-error icon@nwp%nvhpc icon_target=cpu serialize_mode=create +eccodes +ocean'
-            )
+        self.Srun(
+            'spack install --show-log-on-error icon@nwp%nvhpc icon_target=cpu serialize_mode=create +eccodes +ocean'
+        )
 
     def test_devbuild_cpu(self):
         # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
         self.Run('git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git')
```

Comment:
note icon-cscs should be changed to icon-nwp (and corrected below), but probably better in a separate PR

Reply:
Yes and there is already a PR for it #511.